File: warmup-excercises/json/country-info/country-info.js
import OpenAI from 'openai';
import { z } from 'zod';
import { zodResponseFormat } from 'openai/helpers/zod';

const client = new OpenAI({
  baseURL: 'https://ai.company.internal/v1',
  apiKey: 'not-needed'
});

const CountryCapital = z.object({
  country: z.string(),
  capital: z.string(),
  population: z.number(),
  continent: z.string()
});

const response = await client.beta.chat.completions.parse({
  model: 'gpt-4',
  messages: [
    { role: 'user', content: 'Give me information about France' }
  ],
  response_format: zodResponseFormat(CountryCapital, 'country_info')
});

const result = response.choices[0].message.parsed;
console.log(`Country: ${result.country}`);
console.log(`Capital: ${result.capital}`);
console.log(`Population: ${result.population}`);
console.log(`Continent: ${result.continent}`);

File: warmup-excercises/json/country-info/country-info.py
from openai import OpenAI
from pydantic import BaseModel

client = OpenAI(
    base_url="https://ai.company.internal/v1",
    api_key="not-needed"
)

class CountryCapital(BaseModel):
    country: str
    capital: str
    population: int
    continent: str

response = client.beta.chat.completions.parse(
    model="gpt-4",
    messages=[
        {"role": "user", "content": "Give me information about France"}
    ],
    response_format=CountryCapital
)

result = response.choices[0].message.parsed
print(f"Country: {result.country}")
print(f"Capital: {result.capital}")
print(f"Population: {result.population}")
print(f"Continent: {result.continent}")